tox.ini

```diff
@@ -67,6 +67,7 @@ deps =
     py{27,34}-{selects,poll,epolls}: pyopenssl==17.3.0
     setuptools==38.5.1
     {selects,poll,epolls}: pyzmq==17.0.0
+    objgraph==3.4.0
```

Comment:
This is an unnecessarily heavy dependency for what's being done here, we can count the amount of objects of a specific type just fine without it.

Reply:
Will be great to know which method can replace the objgraph.count('eventlet.greenthread.GreenThread')

Reply:
```python
>>> import gc
>>> import objgraph
>>> 
>>> class X:
  2     pass
>>> 
>>> objgraph.count('X')
0
>>> sum(1 for o in gc.get_objects() if isinstance(o, X))
0
>>> x = X()
>>> objgraph.count('X')
1
>>> sum(1 for o in gc.get_objects() if isinstance(o, X))
1
```

Reply:
OK, thanks,
I'll keep it updated with a solution to overcome/change the use of self._exc (if found of-course)

Reply:
it might should be a method of the tests/__init__.py, and it has already gc imported
```python 
def check_ref_count(Instance, expected=0):
    c = sum(1 for o in gc.get_objects() if isinstance(o, Instance))
    assert c == expected, \
        "%s ref expected: %d counted: %d " % (Instance, expected, c)
```

Reply:
or it can be better with a `with statement`
```python 
def check_ref_count(Instance, expected=0):
    try:
        pre_c = sum(1 for o in gc.get_objects() if isinstance(o, Instance))    
    finally:
        c = sum(1 for o in gc.get_objects() if isinstance(o, Instance))
        assert c-pre_c == expected, \
            "%s ref pre-counted: %d expected: %d \ counted: %d "
             % (Instance, pre_c, expected, c)
```
and work with:
```python 
with check_ref_count(Instance):
    #code 
```